backend/backend/core/agents.py

```diff
@@ -28,12 +33,13 @@ def __init__(self, model, python_tool):
         super().__init__()
         prompt = ChatPromptTemplate.from_template(
             """
-            Answer the user's query, using the python_repl tool if needed.
-            Don't include python code in your answer.
-            Instead, execute it using the python_repl tool using a tool call.
-            Make sure to include the result in your final answer.
-            The user query is: {query}
-            """)
+Answer the user's query, using the python_repl tool if needed.
+Don't include python code in your answer.
+Instead, execute it using the python_repl tool using a tool call.
```

Comment:
Instead of executing the code directly, use the python_repl tool for execution and include the result in your final answer.